src/fides/api/ops/util/connection_type.py

```diff
@@ -70,14 +73,65 @@ def connection_type_secret_schema(*, connection_type: str) -> dict[str, Any]:
 
 
 def get_connection_types(
-    search: str | None = None, system_type: SystemType | None = None
+    search: str | None = None,
+    system_type: SystemType | None = None,
+    action_types: Set[ActionType] = SUPPORTED_ACTION_TYPES,
 ) -> list[ConnectionSystemTypeMap]:
     def is_match(elem: str) -> bool:
         """If a search query param was included, is it a substring of an available connector type?"""
         return search.lower() in elem.lower() if search else True
 
+    def saas_request_type_filter(connection_type: str) -> bool:
+        """
+        If any of the request type filters are set to true,
+        ensure the given saas connector supports requests of at least one of those types.
+        """
+        if SUPPORTED_ACTION_TYPES == action_types:
+            # if none of our filters are enabled, pass quickly to avoid unnecessary overhead
+            return True
+
+        template = ConnectorRegistry.get_connector_template(connection_type)
+        if template is None:  # shouldn't happen, but we can be safe
+            return False
+
+        saas_config = SaaSConfig(**yaml.safe_load(template.config).get("saas_config"))
+        has_access = bool(
+            next(
+                (
+                    request.read
+                    for request in [
+                        endpoint.requests for endpoint in saas_config.endpoints
+                    ]
+                ),
+                None,
+            )
+        )
+        has_erasure = (
+            bool(
+                next(
+                    (
+                        request.update or request.delete
+                        for request in [
+                            endpoint.requests for endpoint in saas_config.endpoints
+                        ]
+                    ),
+                    None,
+                )
+            )
+            or saas_config.data_protection_request
+        )
+        has_consent = saas_config.consent_requests
+
+        return bool(
+            (ActionType.consent in action_types and has_consent)
+            or (ActionType.access in action_types and has_access)
+            or (ActionType.erasure in action_types and has_erasure)
+        )
+
     connection_system_types: list[ConnectionSystemTypeMap] = []
-    if system_type == SystemType.database or system_type is None:
+    if (system_type == SystemType.database or system_type is None) and (
+        ActionType.access in action_types or ActionType.erasure in action_types
+    ):
```

Comment:
I believe the dynamo db connector that @SteveDMurphy is working on is going to be our first "consent" db connector.  There may be nothing to do here, just surfacing this is an area that may become outdated.

Reply:
great callout. @SteveDMurphy can we be sure to adjust some of the logic here as you get the dynamo db connector into shape? it shouldn't be too tricky, though the logic in this function is a bit hard to grok, so i'm happy to help out when you get to that point.

i suppose i _could_ try to build some scaffolding to account for consent db connectors here preemptively, but it just feels a bit like over-engineering against an somewhat unknown target. so i think the best thing will be for us to just be diligent as we add that (and any other) db connectors in, for the time being.

Reply:
I agree, this looks great for now